carma_wm/src/WMListener.cpp

```diff
@@ -85,25 +87,37 @@ WMListener::WMListener(
     map_options.callback_group = map_cb_group;
 
     auto route_cb_group = node_base_->create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
-    route_options.callback_group = route_cb_group;                               
+    route_options.callback_group = route_cb_group;
 
     auto roadway_objects_group = node_base_->create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
-    roadway_objects_options.callback_group = roadway_objects_group;                                      
+    roadway_objects_options.callback_group = roadway_objects_group;
 
     auto traffic_spat_group = node_base_->create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
-    traffic_spat_options.callback_group = traffic_spat_group;  
+    traffic_spat_options.callback_group = traffic_spat_group;
 
-  }  
+    auto ros1_clock_group = node_base_->create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
+    ros1_clock_options.callback_group = ros1_clock_group;
+
+    auto sim_clock_group = node_base_->create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
+    sim_clock_options.callback_group = sim_clock_group;
+
+  }
 
   // Setup subscribers
-  route_sub_ = rclcpp::create_subscription<carma_planning_msgs::msg::Route>(node_topics_, "route", 1, 
+  route_sub_ = rclcpp::create_subscription<carma_planning_msgs::msg::Route>(node_topics_, "route", 1,
                                   std::bind(&WMListenerWorker::routeCallback, worker_.get(), std::placeholders::_1), route_options);
 
+  ros1_clock_sub_ = rclcpp::create_subscription<rosgraph_msgs::msg::Clock>(node_topics_, "/clock", 1,
+                                  std::bind(&WMListenerWorker::ros1ClockCallback, worker_.get(), std::placeholders::_1), ros1_clock_options);
+
+  sim_clock_sub_ = rclcpp::create_subscription<rosgraph_msgs::msg::Clock>(node_topics_, "/sim_clock", 1,
+                                  std::bind(&WMListenerWorker::simClockCallback, worker_.get(), std::placeholders::_1), sim_clock_options);
+
```

Comment:
These should be lambdas instead of `std::bind`. Lambdas can completely replace `std::bind` and is the recommended implementation pattern. Consider:
```cpp
ros1_clock_sub_ = rclcpp::create_subscription<rosgraph_msgs::msg::Clock>(node_topics_, "/clock", 1,
[this](const rosgraph_msgs::msg::Clock::SharedPtr msg) {
  this->worker_.simClockCallback(msg);
},
ros1_clock_options);
```

Reply:
This is an optional suggestion, but it would be good to remove `std::bind` usage throughout this file.